### ISSUE TITLE
Intercept mistaken commands (LOAD vs SAVE)

### ISSUE THREAD

**Issue by factus10** [timex-sinclair-projects/TS-Pico#25]
check whether a LOAD "tpi:... " command belongs to the SAVE "tpi:... " commands list

**Comment by rcalcagno** [comment]
Fixed in (upcoming) v1.1c. Please verify and confirm

https://github.com/user-attachments/assets/ece8bfd6-f230-4d1f-82a0-8e9d6f73a1da

**Comment by rcalcagno** [comment]
David, in case you want to know how to test the wrong LOAD "tpi: cd a-m" command with the upcoming v1.1c, here's what would happen:


https://github.com/user-attachments/assets/b0654184-19aa-4939-9f74-3b95202ed27a

**Comment by rkindig** [comment]
I like this!!

Since CD is a command you can only do from SAVE, is there any way to automatically convert it to SAVE and run it?  Or allow the LOAD syntax for the CD command (LOAD makes way more sense to me for CD than SAVE).  I’m sure there are technical reasons it’s part of the SAVE syntax.

Randy

**Comment by factus10** [comment]
Very nice, Ricardo!On Aug 24, 2024, at 10:59 AM, rkindig ***@***.***> wrote:﻿
I like this!!



Since CD is a command you can only do from SAVE, is there any way to automatically convert it to SAVE and run it?  Or allow the LOAD syntax for the CD command (LOAD makes way more sense to me for CD than SAVE).  I’m sure there are technical reasons it’s part of the SAVE syntax.



Randy

**Comment by rcalcagno** [comment]
Well, while this is technically doable, there's a couple reasons it is not advisable:
1) This could interefere with Gustavo's ROM workings, either now or on the future. I don't recall the exact issue, but might be linked to the fact that the current is a "half" implementation of the full specification. For instance, this has no full handshake. So, future versions could implement that and crash. But we can ask the man himself next Wed!
2) This could be (even more) confusing to the final users.

So I don't think it's advisable to add this mod to the main branch, and have the users to enter the correct syntax for the commands. That being said, as the tspico.py will be open starting from this release, it's trivial to add this mod and see if the users find this more suitable, so make those changes permaments. 

Let's add this to the open issues (if there's such thing)

**Comment by ryangray** [comment]
Obviously, there might be ROM reasons preventing it, but a thought I had that might make things easier to remember is if you could move the `dir`, `tapdir`, and `path` commands to `SAVE`, then `LOAD` would only be for mounting, which should be easy to remember. 

I think it's pretty easy now to just remember the few load commands and that all others are SAVE, but it is easy to mis-type it in a hurry. but then I also mis-type LOAD with LIST a lot too.
